pkg/controller/lookup_cache.go

```diff
@@ -52,13 +52,6 @@ type MatchingCache struct {
 	cache *lru.Cache
 }
 
-// NewMatchingCache return a NewMatchingCache, which save label and selector matching relationship.
-func NewMatchingCache(maxCacheEntries int) *MatchingCache {
```

Comment:
Do we need any of the ` MatchingCache` type?  It seems not, by grepping.  I think you can kill off the type and the methods.